templating/base_render_options.go

```diff
@@ -112,14 +114,22 @@ func LoadAndRenderSteps(template *Template, opts *BaseRenderOptions) (string, er
 	return rendered[template.Name], nil
 }
 
-func combineVals(values []string) (string, error) {
+// parseValues receives a slice of values in key=val format
+// and serializes them into JSON. If a key is specified more
+// than once, the key will be overridden.
+func parseValues(values []string) (string, error) {
 	ret := Values{}
 	for _, v := range values {
-		s := strings.Split(v, "=")
-		if len(s) != 2 {
-			return "", fmt.Errorf("failed to parse --set data: %s", v)
+		i := strings.Index(v, "=")
+		if i < 0 {
+			return "", errors.New("failed to parse --set data; invalid format, no = assignment found")
+		}
+		key := v[:i]
+		if key == "" {
+			return "", errors.New("failed to parse --set data; invalid format, empty key")
```

Comment:
empty key => better to tell what is the expected format, eg "--set key=value"?

Reply:
SGTM